src/server/frontend_wayland/session_lock_v1.cpp

```diff
@@ -157,9 +201,64 @@ void mf::SessionLockV1::get_lock_surface(wl_resource* id, wl_resource* surface_r
             std::to_string(wl_resource_get_id(output)) +
             " does not map to a valid Mir output ID"));
     }
+
     new SessionLockSurfaceV1{id, manager, WlSurface::from(surface_resource), output_id.value()};
 }
 
+void mf::SessionLockV1::destroy()
+{
+    // Note that this is a deviation from the spec. The spec reads:
+    // "It is a protocol error to make this request if the locked event was sent,
+    // the unlock_and_destroy request must be used instead."
+    //
+    // In our case, it is ONLY a protocol error if the session is still locked,
+    // as the session may have been unlocked via logind independent of the
+    // wayland protocol.
+
+    if (manager.surface_stack->screen_is_locked())
```

Comment:
In the new era `SessionLocker` should we be asking the `surface_stack` about locks?